src/test/java/com/meilisearch/integration/TenantTokenTest.java

```diff
@@ -0,0 +1,244 @@
+        TenantTokenOptions options = new TenantTokenOptions();
+        options.setExpiresAt(Date.from(tomorrow));
+
+        String jwtToken = privateClient.generateTenantToken(rules, options);
+
+        Client tokenClient = new Client(new Config("http://localhost:7700", jwtToken));
+
+        SearchResult searchResult = tokenClient.index(indexUid).search("");
+
+        assertEquals(0, searchResult.getHits().size());
+        assertEquals(20, searchResult.getLimit());
+    }
+
+    /** Test Create Tenant Token with api key */
+    @Test
+    public void testGenerateTenantTokenWithApiKey() throws Exception {
+        String indexUid = "GenerateTokenWithApiKey";
+        Index index = createEmptyIndex(indexUid);
+        Key key = getPrivateKey();
+
+        Map<String, Object> rules = new HashMap<String, Object>();
+        rules.put("*", new HashMap<String, Object>());
+
+        TenantTokenOptions options = new TenantTokenOptions();
+        options.setApiKey(key.getKey());
+
+        String jwtToken = client.generateTenantToken(rules, options);
+
+        Client tokenClient = new Client(new Config("http://localhost:7700", jwtToken));
+
+        SearchResult searchResult = tokenClient.index(indexUid).search("");
+
+        assertEquals(0, searchResult.getHits().size());
+        assertEquals(20, searchResult.getLimit());
+    }
+
+    /** Test Create Tenant Token with all options */
+    @Test
+    public void testGenerateTenantTokenWithAllOptions() throws Exception {
+        String indexUid = "GenerateTokenWithAllOptions";
+        Index index = createEmptyIndex(indexUid);
+        Key key = getPrivateKey();
+
+        Map<String, Object> rules = new HashMap<String, Object>();
+        rules.put("*", new HashMap<String, Object>());
+
+        Instant instant = Instant.now();
+        Instant tomorrow = instant.plus(1, ChronoUnit.DAYS);
+        TenantTokenOptions options = new TenantTokenOptions();
+        options.setApiKey(key.getKey());
+        options.setExpiresAt(Date.from(tomorrow));
+
+        String jwtToken = client.generateTenantToken(rules, options);
+
+        Client tokenClient = new Client(new Config("http://localhost:7700", jwtToken));
+
+        SearchResult searchResult = tokenClient.index(indexUid).search("");
+
+        assertEquals(0, searchResult.getHits().size());
+        assertEquals(20, searchResult.getLimit());
+    }
+
+    /** Test Create Tenant Token with no search rules */
+    @Test
+    public void testGenerateTenantTokenWithNoSearchRules() throws Exception {
+        String indexUid = "GenerateTokenWithNoSearchRules";
+        Key key = getPrivateKey();
+
+        Client privateClient = new Client(new Config("http://localhost:7700", key.getKey()));
+
+        assertThrows(MeiliSearchException.class, () -> privateClient.generateTenantToken(null));
+    }
+
+    /** Test Create Tenant Token with bad expireation date */
+    @Test
+    public void testGenerateTenantTokenWithBadExpiresAt() throws Exception {
```

Comment:
Could you add a test about UTC as well?

Reply:
I'm not sure to know what you think about it?

Reply:
I've added validations to expiresAt to check if they come as UTC value both in dart and ruby, I don't know if this is overkill (because the user can have problems with that).

Reply:
Alright, get it. Yes, I'm not really comfortable adding this verification I'm afraid it could create errors on the user side depending on how he prefers to use this function. I think a note in the definition of the function is enough no?